spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -164,6 +177,22 @@ private boolean useHiveContext() {
     return Boolean.parseBoolean(getProperty("zeppelin.spark.useHiveContext"));
   }
 
+  public boolean diagnosis() {
+    if (getProperty("zeppelin.spark.diagnosis") == null) {
+      return false;
+    } else {
+      return Boolean.parseBoolean(getProperty("zeppelin.spark.diagnosis"));
+    }
+  }
+
+  public boolean isYarnMode() {
+    return getProperty("master").equals("yarn-client");
```

Comment:
essentially, here

when deployMode is not set but master starts with "yarn" it gets default to yarn-client
```
case (_, mode) =>
          args.master = "yarn-" + Option(mode).getOrElse("client")
```